Documentation/troubleshooting.rst

```diff
@@ -0,0 +1,203 @@
+.. _admin_guide:
+
+###############
+Troubleshooting
+###############
+
+This document describes how to troubleshoot Cilium in different deployment
+modes. It focuses on a full deployment of Cilium within a datacenter or public
+cloud. If you are just looking for a simple way to experiment, we highly
+recommend trying out the :ref:`gs_guide` instead.
+
+This guide assumes that you have read the :ref:`arch_guide` which explains all
+the components and concepts.
+
+Kubernetes
+==========
+
+Check the status of the DaemonSet_ and verify that all desired instances are in
+"ready" state:
+
+.. code:: bash
+
+        $ kubectl --namespace kube-system get ds
+        NAME      DESIRED   CURRENT   READY     NODE-SELECTOR   AGE
+        cilium    1         1         0         <none>          3s
+
+In this example, we see a desired state of 1 with 0 being ready. This indicates
+a problem. The next step is to list all cilium pods by matching on the label
+``k8s-app=cilium`` and also sort the list by the restart count of each pod to
+easily identify the failing pods:
+
+.. code:: bash
+
+        $ kubectl --namespace kube-system get pods --selector k8s-app=cilium \
+                  --sort-by='.status.containerStatuses[0].restartCount'
+        NAME           READY     STATUS             RESTARTS   AGE
+        cilium-813gf   0/1       CrashLoopBackOff   2          44s
+
+Pod ``cilium-813gf`` is failing and has already been restarted 2 times. Let's
+print the logfile of that pod to investigate the cause:
+
+.. code:: bash
+
+        $ kubectl --namespace kube-system logs cilium-813gf
+        INFO      _ _ _
+        INFO  ___|_| |_|_ _ _____
+        INFO |  _| | | | | |     |
+        INFO |___|_|_|_|___|_|_|_|
+        INFO Cilium 0.8.90 f022e2f Thu, 27 Apr 2017 23:17:56 -0700 go version go1.7.5 linux/amd64
+        CRIT kernel version: NOT OK: minimal supported kernel version is >= 4.8
+
+In this example, the cause for the failure is a Linux kernel running on the
+worker node which is not meeting :ref:`admin_system_reqs`.
+
+If the cause for the problem is not apparent based on these simple steps,
+please come and seek help on our `Slack channel`_.
+
+Testing External Connectivity
+=============================
+
+IPv6 external connectivity can be tested with:
```

Comment:
I'm curious why this jumps right to v6 connectivity testing, without showing IPv4.  Is the assumption that this is obvious, but that there are tricky issues with testing of v6?